src/libraries/System.Linq/src/System/Linq/Sum.cs

```diff
@@ -40,6 +43,26 @@ public static partial class Enumerable
             where T : struct, INumber<T>
             where TResult : struct, INumber<TResult>
         {
+            if (typeof(T) == typeof(TResult)
+                && Vector<T>.IsSupported
+                && Vector.IsHardwareAccelerated
+                && Vector<T>.Count > 2
+                && span.Length >= Vector<T>.Count * 4)
+            {
+                // For cases where the vector may only contain two elements vectorization doesn't add any benefit
+                // due to the expense of overflow checking. This means that architectures where Vector<T> is 128 bit,
+                // such as ARM or Intel without AVX, will only vectorize spans of ints and not longs.
+
+                if (typeof(T) == typeof(long))
+                {
+                    return (TResult) (object) SumSignedIntegersVectorized(MemoryMarshal.Cast<T, long>(span));
```

Comment:
If `T` equals `long`, why would you need to even cast the span? Seems calling the helper method with `T` as the generic parameter would work just fine.

Reply:
The cast is necessary because the generic math within vectorization requires `IBinaryInteger<T>`, which isn't present since this method also supports floating point generics.

Reply:
Correct